### PR TITLE
Document usages of the Int32EntityId interface

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Resubmission of https://github.com/wmde/WikibaseDataModel/pull/674 by @thiemowmde

Please regard this as if it where the original PR. I accidentally merged it, and then reverted plus resubmitted to undo my mistake. This should have no bearing on further review.

### CODE REVIEW

File: src/Entity/ItemId.php [hunk 1]
```diff
@@ -45,7 +45,9 @@ private function assertValidIdFormat( $idSerialization ) {
 	}
 
 	/**
-	 * @return int
+	 * @see Int32EntityId::getNumericId
+	 *
+	 * @return int Guaranteed to be a unique integer in the range [1..2147483647].
```
Comment (JeroenDeDauw) [inline src/Entity/ItemId.php]: Don't think the "unique" part is correct. Or at least needs some very specific interpretation that is not specified. You can clearly have multiple instances with the same value. The uniqueness is only there in some context. (Which is why this class does not say anything about this in the first place)

Comment (thiemowmde) [inline src/Entity/ItemId.php]: See https://github.com/wmde/WikibaseDataModel/pull/674#discussion_r77503685.

**Comment by brightbyte** [comment]
what about this?

**Comment by thiemowmde** [comment]
I don't know. Help me understand what I should do to get this merged? Or just merge it?

**Comment by brightbyte** [comment]
Seems to need a rebase now.

**Comment by thiemowmde** [comment]
Rebased and updated to use "distinct" instead of "unique".

**Comment by JeroenDeDauw** [comment]
I think `distinct` has the same problem as `unique`.

**Comment by thiemowmde** [comment]
In https://github.com/wmde/WikibaseDataModel/pull/677#r77503892 I already explained the purpose and relevance of this line of documentation extensively. The method guarantees to return a different number for every non-equal object. Without this guarantee being documented the method is of not much value. Please provide a way forward.

**Pull request merged**